cmp/report_slices.go

```diff
@@ -128,8 +127,10 @@ func (opts formatOptions) FormatDiffSlice(v *valueNode) textNode {
 				numLines++
 			}
 		}
-		isText = !isBinary
-		isLinedText = isText && numLines >= 4 && maxLineLen <= 1024
+		isPureText := numValidRunes == numTotalRunes
+		isMostlyText = float64(numValidRunes)/float64(numTotalRunes) > 0.95
```

Comment:
ISTM that this only triggers on longer strings. For example, if there's one non-printable character in a 19 character string, it won't be treated as "mostly text". Perhaps there could be some kind of count cut-off too?

Reply:
The challenged with the heuristics is that they're never right 😞.
The benefit with heuristics is that they don't need to be right and we can do whatever is best for humans 😏.

> For example, if there's one non-printable character in a 19 character string, it won't be treated as "mostly text".

Did you mean something like?
```go
isMostlyText = float64(numValidRunes)/float64(numTotalRunes) > 0.95 || numTotalRunes-numValidRunes < 4
```

Reply:
yeah, something like that.

how about this?

```
	const mostlyTextProportion = 0.90

	isMostlyText := numTotalRunes == numValidRunes
	if !isMostlyText && numTotalRunes > 3 {
		threshold := int(math.Floor(mostlyTextProportion * float64(numTotalRunes)))
		isMostlyText = numValidRunes >= threshold
	}
```

That gives the following set of cutoff points (number of invalid characters / total number of characters):

```
0/1
0/2
0/3
1/4
1/5
1/6
1/7
1/8
1/9
1/10
2/11
2/12
2/13
2/14
2/15
2/16
2/17
2/18
2/19
2/20
3/21
3/22
3/23
3/24
3/25
3/26
3/27
3/28
3/29
3/30
4/31
4/32
4/33
4/34
4/35
4/36
4/37
4/38
4/39
4/40
5/41
5/42
5/43
5/44
5/45
5/46
5/47
5/48
5/49
5/50
6/51
6/52
6/53
6/54
6/55
6/56
6/57
6/58
6/59
6/60
7/61
7/62
7/63
7/64
7/65
7/66
7/67
7/68
7/69
7/70
8/71
8/72
8/73
8/74
8/75
8/76
8/77
8/78
8/79
8/80
9/81
9/82
9/83
9/84
9/85
9/86
9/87
9/88
9/89
9/90
10/91
10/92
10/93
10/94
10/95
10/96
10/97
10/98
10/99
```

Reply:
The heuristic seems fairly complicated. Since most of those ratios are below 80%, I think it would be easier and simpler to just reduce the ratio to 80%.

Reply:
https://play.golang.org/p/M6s-3G_Wg-T

Reply:
FWIW the heuristic was "round down from 90%", but I'm happy to go with 80% (which gives these numbers, I think: https://play.golang.org/p/lCIupl-0WMO).

Reply:
I went with math.Floor, but keep the threshold at 90%. We can adjust this in the future as necessary.